src/spells.cpp

```diff
@@ -837,11 +837,11 @@ bool InstantSpell::playerCastInstant(Player* player, std::string& param)
 	if (selfTarget) {
 		var.setNumber(player->getID());
 	} else if (needTarget || casterTargetOrDirection) {
-		Creature* target = nullptr;
+		std::shared_ptr<Creature> target = nullptr;
```

Comment:
Could this be `auto`
```suggestion
		auto target = std::shared_ptr<Creature>(nullptr);
```